### PR TITLE
Diagnostic section

### PR DESCRIPTION
**Pull request by danielmarbach**
```
  "NServiceBus.Persistence.AzureTable.StorageSession": {
    "ConnectionMechanism": "Custom",
    "DefaultTable": "tmp9ee1637406989773864702"
  },
```

and

```
  "NServiceBus.Persistence.AzureTable.Subscriptions": {
    "ConnectionMechanism": "Custom",
    "TableName": "tmp6b44637406990952427456",
    "CacheFor": {
      "Ticks": 0,
      "Days": 0,
      "Hours": 0,
      "Milliseconds": 0,
      "Minutes": 0,
      "Seconds": 0,
      "TotalDays": 0,
      "TotalHours": 0,
      "TotalMilliseconds": 0,
      "TotalMinutes": 0,
      "TotalSeconds": 0
    }
```

### CODE REVIEW

File: src/NServiceBus.Persistence.AzureTable/Subscriptions/SubscriptionStorage.cs [hunk 1]
```diff
@@ -29,6 +28,16 @@ protected override void Setup(FeatureConfigurationContext context)
...
             var subscriptionTableName = context.Settings.GetSubscriptionTableName();
             var cacheFor = context.Settings.GetOrDefault<TimeSpan>(WellKnownConfigurationKeys.SubscriptionStorageCacheFor);
+
+            context.Settings.AddStartupDiagnosticsSection(
+                "NServiceBus.Persistence.AzureTable.Subscriptions",
+                new
+                {
+                    ConnectionMechanism = cloudTableClientProvider is CloudTableClientForSubscriptionsFromConnectionString ? "ConnectionString" : "Custom",
+                    TableName = subscriptionTableName,
+                    CacheFor = cacheFor,
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.AzureTable/Subscriptions/SubscriptionStorage.cs:38]: Should be outputting “default” if that’s used. Also, no matter what we do, writing all TimeSpan fields brings no value to customers.

**Review by SeanFeldman — APPROVED** [review]
Approved With comments

File: src/NServiceBus.Persistence.AzureTable/Subscriptions/SubscriptionStorage.cs (hunk 1 above)
Comment (danielmarbach) [reply src/NServiceBus.Persistence.AzureTable/Subscriptions/SubscriptionStorage.cs:38]: Switched to CloudTableClient instead

**Comment by danielmarbach** [comment]
Also raised https://github.com/Particular/NServiceBus.Persistence.AzureStorage/pull/332 because the other PR was already merged

**Pull request merged**